Source/warningreporting/WarningReportingUnit.cpp

```diff
@@ -279,6 +282,8 @@ namespace WarningReporting {
             fflush(stdout);
         }
 
+        _adminLock.Unlock();
```

Comment:
And I think here we could move the Unlock up. At least to just after the _directOuput == true (but does that even need checking? Can it change at runtime?